### PR TITLE
[WIP] Testing client integration 3

### PR DESCRIPTION
**Pull request by hswick**
This test will act as source of truth to test client functionality. If changes don't pass these tests, they don't work...

getContracts is not functioning as expected and is currently being replaced by passing the truffle contracts into the bridge at a high level.

### CODE REVIEW

**Comment by shrugs** [comment]
well, the tests run and pass, but don't exit; was seeing the same behavior on 76a27c1

File: test/integration/challenger.js [hunk 1]
```diff
@@ -80,24 +80,27 @@ describe('Challenger Client Integration Tests', function () {
...
-        let session = await bridge.api.getSession(sessionId)
-        let step = session.medStep.toNumber()
-        // let highStep = session.highStep.toNumber()
-        // let lowStep = session.lowStep.toNumber()
-
-        let results = await bridge.api.getResult(session.input, step)
-
-        await bridge.api.respond(sessionId, step, results.stateHash, { from: claimant })
-      }
+      await new Promise(async (resolve, reject) => {
```
Comment (shrugs) [inline test/integration/challenger.js]: this shouldn't be necessary

Comment (hswick) [reply test/integration/challenger.js]: The loop? You have to go back and forth between claimant and challenger

Comment (shrugs) [reply test/integration/challenger.js]: The loop is fine, you don't need to wrap it in a promise like that. Why was this added?

**Pull request merged**